homeassistant/components/recollect_waste/__init__.py

```diff
@@ -1 +1,87 @@
-"""The recollect_waste component."""
+"""The Recollect Waste integration."""
+import asyncio
+from datetime import date, timedelta
+from typing import List
+
+from aiorecollect.client import Client, PickupEvent
+from aiorecollect.errors import RecollectError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_PLACE_ID, CONF_SERVICE_ID, DATA_COORDINATOR, DOMAIN, LOGGER
+
+DATA_LISTENER = "listener"
+
+DEFAULT_NAME = "recollect_waste"
+DEFAULT_UPDATE_INTERVAL = timedelta(days=1)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+    """Set up the RainMachine component."""
+    hass.data[DOMAIN] = {DATA_COORDINATOR: {}, DATA_LISTENER: {}}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up RainMachine as config entry."""
+    session = aiohttp_client.async_get_clientsession(hass)
+    client = Client(
+        entry.data[CONF_PLACE_ID], entry.data[CONF_SERVICE_ID], session=session
+    )
+
+    try:
+        await client.async_get_next_pickup_event()
+    except RecollectError as err:
+        LOGGER.error("Error setting up Recollect sensor platform: %s", err)
+        return False
```

Comment:
If it's connection related, raise ConfigEntryNotReady. For auth errors, return False.
```suggestion
        raise ConfigEntryNotReady
```